### PR TITLE
[2.11.x] DDF-3530 Fixed storage not being properly rolled back on ingest via content directory monitor

### PR DESCRIPTION
**Pull request by Lambeaux**
#### What does this PR do?
Fixed storage not being properly rolled back on ingest via content directory monitor. 

#### Who is reviewing it? 
@brjeter 
@blen-desta  

#### Choose 2 committers to review/merge the PR. 
@bdeining
@coyotesqrl

#### How should this be tested? (List steps with links to updated documentation)
TBD

#### Any background context you want to provide?
TBD

#### What are the relevant tickets?
[DDF-3530](https://codice.atlassian.net/browse/DDF-3530)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/IngestException.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 /**
  * The Class {@link IngestException} is thrown by a {@link CatalogProvider} when an error occurs on
  * an operation.
+ *
```
Comment (brjeter) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/IngestException.java]: 😂

Comment (Lambeaux) [reply catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/IngestException.java]: Yep. Was going to pre-comment this but you beat me to it :) 

❓ Should we add javadoc explaining if `IngestException` implies complete failure? Or partial failure?

**Comment by Lambeaux** [comment]
Build now.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java [hunk 2]
```diff
@@ -202,6 +198,17 @@ public CreateResponse create(
     return createResponse;
   }
 
+  private void rollbackStorage(CreateStorageRequest createStorageRequest) {
+    if (createStorageRequest != null) {
+      try {
+        sourceOperations.getStorage().rollback(createStorageRequest);
+      } catch (StorageException e1) {
+        LOGGER.info(
```
Comment (bdeining) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java]: Should this be logged at the `info` level?

Comment (Lambeaux) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java]: I grabbed existing code and pulled it into its own method.

**Comment by Lambeaux** [comment]
Build now.

**Comment by Lambeaux** [comment]
Due to scope of this PR and the magnitude of standard framework SonarQube issues, we decided to disregard them for now.

**Comment by brjeter** [comment]
Hero passed ✅ 
 - Ran itests of DDF and downstream projects
 - Manually verified the fix

**Pull request merged**